pymatviz/ptable.py

```diff
@@ -1174,11 +1175,11 @@ def ptable_plots(
                 scatter_kwargs = plot_kwds or {}
 
             # Plot without colormap when data len is 2
-            if plot_data.ndim == 2:
+            if len(plot_data) == 2:
                 ax.plot(plot_data[0], plot_data[1], **(scatter_kwargs or {}))
 
             # Plot with colormap when data len is 3
-            elif plot_data.ndim == 3:
+            elif len(plot_data) == 3:
```

Comment:
I'm a little confused what we were trying to achieve here:
```python
            # Plot with colormap when data len is 3
            elif len(plot_data) == 3:
                ax.plot(
                    plot_data[0],
                    plot_data[1],
                    c=plot_data[2],
                    cmap=cmap,
                    **(scatter_kwargs or {}),
                )
```
The `plot_data[2]` should be a array the same length as `plot_data[0/1]`. But I think we could only assign a single color for a line (unlike the case for scatters?)?

Reply:
yes, this is the issue i noticed in https://github.com/janosh/pymatviz/issues/132#issuecomment-2049115252. specifically what i meant when i wrote

> this should be `if len(plot_data) == 2:` and `if len(plot_data) == 3:` instead of checking `ndim`. 🤦
> 
> the annoying thing is, `ax.plot()` only supports a `color` (or `c`) as a single string, not different colors as a list of values for each data point. so we need to go back to `ax.scatter` for the `len=3` case. but then the `scatter_kwargs` are broken since `ax.plot` and `ax.scatter` have very different keywords. quite annoying.

i don't have a good solution yet. i briefly tried changing the color of each data point afterwards like this

```py
for point, color in zip(zip(*plot_data[:2]), plot_data[2]):
    ax.plot(*point, color=cmap(color), **scatter_kwargs)

```

but don't think this will work and would be a bad solution anyway

Reply:
I would fold this for now because I'm confident this problem would be gone if we could manage to achieve #133 😃